pkg/types/types.go

```diff
@@ -75,54 +75,115 @@ const (
+type RuntimeName string
 
-	// Pod namespace where the event comes from, or empty for host-level
-	// event
-	Namespace string `json:"namespace,omitempty" column:"namespace,template:namespace" columnTags:"kubernetes"`
+func (r RuntimeName) String() string {
+	return string(r)
+}
 
-	// Pod where the event comes from, or empty for host-level event
-	Pod string `json:"pod,omitempty" column:"pod,template:pod" columnTags:"kubernetes"`
+const (
+	RuntimeNameDocker     RuntimeName = "docker"
+	RuntimeNameContainerd RuntimeName = "containerd"
+	RuntimeNameCrio       RuntimeName = "cri-o"
+	RuntimeNamePodman     RuntimeName = "podman"
+	RuntimeNameUnknown    RuntimeName = "unknown"
+)
 
-	// Container where the event comes from, or empty for host-level or
-	// pod-level event
-	Container string `json:"container,omitempty" column:"container,template:container" columnTags:"kubernetes,runtime"`
+func String2RuntimeName(name string) RuntimeName {
+	switch name {
+	case string(RuntimeNameDocker):
+		return RuntimeNameDocker
+	case string(RuntimeNameContainerd):
+		return RuntimeNameContainerd
+	case string(RuntimeNameCrio):
+		return RuntimeNameCrio
+	case string(RuntimeNamePodman):
+		return RuntimeNamePodman
+	}
+	return RuntimeNameUnknown
+}
+
+type BasicRuntimeMetadata struct {
+	// RuntimeName is the name of the container runtime. It is useful to distinguish
+	// who is the "owner" of each container in a list of containers collected
+	// from multiple runtimes.
+	RuntimeName RuntimeName `json:"runtimeName,omitempty" column:"runtimeName,minWidth:5,maxWidth:12"`
+
+	// ContainerID is the container ContainerID without the container runtime prefix. For
+	// instance, without the "cri-o://" for CRI-O.
+	ContainerID string `json:"containerId,omitempty" column:"containerId,width:13,maxWidth:64"`
+
+	// ContainerName is the container name. In the case the container runtime
+	// response with multiple containers, ContainerName contains only the first element.
+	ContainerName string `json:"containerName,omitempty" column:"containerName,template:container"`
+}
+
+func (b *BasicRuntimeMetadata) IsEnriched() bool {
+	return b.RuntimeName != RuntimeNameUnknown && b.RuntimeName != "" && b.ContainerID != "" && b.ContainerName != ""
+}
+
+type BasicK8sMetadata struct {
+	Namespace     string `json:"namespace,omitempty" column:"namespace,template:namespace"`
+	PodName       string `json:"podName,omitempty" column:"pod,template:pod"`
+	ContainerName string `json:"containerName,omitempty" column:"container,template:container"`
+}
+
+func (b *BasicK8sMetadata) IsEnriched() bool {
+	return b.Namespace != "" && b.PodName != "" && b.ContainerName != ""
+}
+
+type K8sMetadata struct {
+	Node string `json:"node,omitempty" column:"node,template:node"`
+
+	BasicK8sMetadata `json:",inline"`
 
 	// HostNetwork is true if the container uses the host network namespace
 	HostNetwork bool `json:"hostNetwork,omitempty" column:"hostnetwork,hide"`
 }
 
+type CommonData struct {
```

Comment:
That's in the TODO list for future PRs because it is tricky to enable `runtime` columns in `kubectl-gadget` without showing them by default as it is in `ig`. Vice versa in `ig` for `k8s` columns. In addition, I was considering adding a flag to display such info `ig <gadget> --k8s` and `kubectl-gadget <gadget> --runtime`. WDYT?